src/gluonts/dataset/repository/_zenodo.py

```diff
@@ -105,17 +134,29 @@ def save_metadata(
         )
 
 
-def save_dataset(dataset_path: Path, data: List[Dict]):
-    save_to_file(
-        dataset_path / "data.json",
-        [
-            to_dict(
+def save_datasets(path: Path, data: List[Dict], train_offset: int):
+    train = path / "train"
+    train.mkdir(exist_ok=True)
+    test = path / "test"
+    test.mkdir(exist_ok=True)
+
+    with open(train / "data.json", "wb") as train_fp, open(
+        test / "data.json", "wb"
+    ) as test_fp:
+        for data_entry in tqdm(
+            data, total=len(data), desc="creating json files"
+        ):
+            dic = to_dict(
                 target_values=data_entry["target"],
                 start=str(data_entry["start_timestamp"]),
             )
-            for data_entry in data
-        ],
-    )
+
+            test_fp.write(json.dumps(dic).encode("utf-8"))
+            test_fp.write("\n".encode("utf-8"))
+
+            dic["target"] = dic["target"][:-train_offset]
+            train_fp.write(json.dumps(dic).encode("utf-8"))
+            train_fp.write("\n".encode("utf-8"))
```

Comment:
I know that you don't like it, but I benchmarked it and it's nearly twice as fast as the previous implementation because we iterate over the data only once. Also it makes sense to create the "train/data.json"  after the "test/data.json", so we don't have to save it twice, but I am open for a different approach if you have one

Reply:
What is it that I don't like?

I think it should be able to open the files with `utf-8` encoding, that should spare you from writing `.encode`.